AIAgent/ml/validation/coverage/game_managers/model/process_game_manager.py

```diff
@@ -0,0 +1,330 @@
+                or len(self._games_info[map_name].total_steps) == 0
+            ):
+                logger = logging.error
+            else:
+                logger = logging.debug
+            self._get_and_log_proc_output(proc, logger)
+        except FunctionTimedOut as error:
+            logging.warning(f"Timeouted on map {map_name} with {error.timedOutAfter}s")
+            self._kill_game_process(proc, logging.warning)
+            game_result = GameFailed(reason=type(error))
+        except Exception as e:
+            logging.error(e, exc_info=True)
+            if running_res is not None:
+                self._kill_game_process(proc, logging.error)
+            game_result = GameFailed(reason=type(e))
+        torch.cuda.empty_cache()
+        return Map2Result(game_map2svm, game_result)
+
+    def _get_output_dir(self, game_map: GameMap) -> Path:
+        return SVMS_OUTPUT_PATH / f"{game_map.MapName}"
+
+    def _run_game_process(self, game_map2svm: GameMap2SVM):
+        game_map, svm_info = game_map2svm.GameMap, game_map2svm.SVMInfo
+        svm_info = game_map2svm.SVMInfo
+
+        def look_for_free_port(attempts=100) -> int:
+            if attempts <= 0:
+                raise RuntimeError("Failed to occupy port")
+            logging.debug(f"Looking for port... Attempls left: {attempts}.")
+            try:
+                with self._shared_lock:
+                    port = next_free_port(svm_info.min_port, svm_info.max_port)
+                    logging.debug(f"Try to occupy {port=}")
+                    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                    server_socket.bind(
+                        ("localhost", port)
+                    )  # TODO: working within a local network
+                    server_socket.listen(1)
+                    return port, server_socket
+            except OSError:
+                logging.debug("Failed to occupy port")
+                return look_for_free_port(attempts - 1)
+
+        port, server_socket = look_for_free_port()
+        launch_command = svm_info.launch_command.format(
+            **{
+                STEPS_TO_PLAY: game_map.StepsToPlay,
+                DEFAULT_SEARCHER: game_map.DefaultSearcher,
+                STEPS_TO_START: game_map.StepsToStart,
+                ASSEMBLY_FULL_NAME: game_map.AssemblyFullName,
+                NAME_OF_OBJECT_TO_COVER: game_map.NameOfObjectToCover,
+                MAP_NAME: game_map.MapName,
+                MODEL_PATH: Path(CURRENT_ONNX_MODEL_PATH).absolute(),
+                OUTPUT_DIR: self._get_output_dir(game_map2svm.GameMap).absolute(),
+                PORT: port,
+            },
+        )
+        logging.info(f"Launch command: {launch_command}")
+        proc = subprocess.Popen(
+            launch_command.split(),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            cwd=Path(svm_info.server_working_dir),
```

Comment:
`Path(svm_info.server_working_dir).resolve`? If user provides relative path, this can fail to find the correct wdir